### PR TITLE
Build dep resolver and ldflags

### PR DESCRIPTION
**Pull request by dorileo**
This patchset changes the dependency-resolver.py to follow what @cmarcelo, @lucasdemarchi and I discussed yesterday, also, adds the missing ldflags support.

@ulissesf could you check where we should set the -extra-ldflags for pthread since you've touched these codes?

### CODE REVIEW

**Comment by dorileo** [comment]
look, this patchset brings a patch from @ulissesf which is already here https://github.com/solettaproject/soletta/pull/141

**Comment by dorileo** [comment]
I'm checking why this is failing on ubuntu...

**Comment by dorileo** [comment]
ok, now it builds on ubuntu :)

File: data/jsons/dependencies.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	},
+	{
+	    "dependency": "systemd",
+	    "type": "pkg-config",
+	    "pkgname": "libsystemd"
+	},
+	{
+	    "dependency": "udev",
+	    "type": "pkg-config",
+	    "pkgname": "libudev"
```
Comment (lpereira) [inline data/jsons/dependencies.json]: Shouldn't the versions of pkg-config things be checked as well?

Comment (dorileo) [inline data/jsons/dependencies.json]: do you mean with --modversion ?

Comment (dorileo) [inline data/jsons/dependencies.json]: Do you think it's worth it? maybe for some cases, right?

**Comment by dorileo** [comment]
just sent new version, https://github.com/solettaproject/soletta/pull/143
closing this one
@lpereira this v2 I've just sent doesn't implement any version check.

**Pull request closed without merging**